### ISSUE TITLE
libuv Windows : uv_fs_init changes default file reading behavior of the current process.

### ISSUE THREAD

**Issue by GregGodin** [libuv/libuv#840]
It changes the _fmode global variable, and thus, in windows, it changes the behavior of std::ifstream.
Reading text like this :

```
std::ifstream stream("C:\\Test.txt", std::ifstream::in);
std::stringstream str;
str << stream.rdbuf();
```

produce stuff with \n carriage returns before the call uv_fs_init, and gives stuff with \r\n after the call of uv_fs_init (the ifstream doesn't default to text mode anymore).

**Comment by piscisaureus** [comment]
Agree that the _fmode change has no place in libuv. It's also superfluous, since libuv doesn't use CRT i/o functions. So go ahead, and kill uv_fs_init().

Unfortunately this is an ABI change, so we can't release it until libuv 2.0 drops, whenever that is.
E.g. I'm pretty sure that node.js would be affected in certain edge cases (e.g. certain warnings and errors are written using `write(2, ...)`, so people's log files will suddenly contain windows-style line endings.

**Comment by saghul** [comment]
Hum, is it really an ABI change from libuv's prespective? Node could continue to do what libuv did in a next semver-major and would just find libuv doing The Right Thing?

**Comment by piscisaureus** [comment]
@saghul Yes, what you said.

**Comment by vtjnash** [comment]
Implemented in #1166.

**Comment by cjihrig** [comment]
Closing as done. It will be in libuv 2.0.0
